orchestra/accounts/views.py

```diff
@@ -147,18 +157,79 @@ def get(self, request, *args, **kwargs):
         return render(request, self.template_name, {
             'form_data': zip(comm_pref_formset, self.descriptions),
             'comm_pref_formset': comm_pref_formset,
+            'worker': self.worker,
         })
 
     def post(self, request, *args, **kwargs):
         comm_pref_formset = self.CommunicationPreferenceFormSet(
             data=request.POST)
         self.set_method_choices(comm_pref_formset)
         success = comm_pref_formset.is_valid()
-
         if success:
             comm_pref_formset.save()
         return render(request, self.template_name, {
             'form_data': zip(comm_pref_formset, self.descriptions),
             'comm_pref_formset': comm_pref_formset,
+            'worker': self.worker,
             'success': success,
         })
+
+
+class AvailabilitySettingsView(WorkerViewMixin):
+    template_name = 'accounts/availability_settings.html'
+    form_class = WorkerAvailabilityForm
+
+    def set_context_data(self, request, *args, **kwargs):
+        super().set_context_data(request, *args, **kwargs)
+        now = timezone.now()
+        self.this_week = first_day_of_the_week(now)
+        self.next_week = first_day_of_the_week(now + timedelta(days=7))
+        this_week_availability = WorkerAvailability.objects.filter(
+            worker=self.worker, week=self.this_week).first()
+        next_week_availability = WorkerAvailability.objects.filter(
+            worker=self.worker, week=self.next_week).first()
+        this_week_prefix = 'this_week'
+        next_week_prefix = 'next_week'
+        self.this_week_form = WorkerAvailabilityForm(
+            data=request.POST or None, prefix=this_week_prefix,
+            instance=this_week_availability)
+        self.next_week_form = WorkerAvailabilityForm(
+            data=request.POST or None, prefix=next_week_prefix,
+            instance=next_week_availability)
+
+    def _render(self, request, **kwargs):
+        kwargs.update({
+            'this_week_availability_form': self.this_week_form,
+            'next_week_availability_form': self.next_week_form,
+            'worker': self.worker,
+        })
+        return render(request, self.template_name, kwargs)
+
+    def _update_form(self, form, week):
+        success = form.is_valid()
+        if success:
+            # Set private fields we didn't send to the frontend
+            form.instance.worker = self.worker
+            form.instance.week = week
+            form.save()
+        return success
+
+    def get(self, request, *args, **kwargs):
+        if self.worker.max_autostaff_hours_per_day <= 0:
```

Comment:
i like being defensive! just sanity-checking that this could only be reached if someone opened the form, saw the tab, had max=0 set, then clicked on the tab. does that sound right?

Reply:
there's also a direct URL to this tab, so this is more than hypothetically defensive.